### PR TITLE
build: add test for failing osbuild and rework streaming

### PR DESCRIPTION
**Pull request by mvo5**
When adding a test that the error handling is working correctly the test would fails every 1/10 time with an unexpected EOF error. Looking deeper it seems the followLineOutput code was racy and also unnecessary. By just using a wrapped writer and passing that to `exec.Command.{Stdout,Stderr}` all the heavy lifting about output handling is done by the exec package. To ensure that with that we still get streamed output a test is added that checks that streaming works.

The test also shows that some tweaks are needed inside the handler (i.e. we cannot use http.Error(), see code comment).

### CODE REVIEW

**Comment by mvo5** [comment]
For some reason I can not add you as a reviewer @croissanne - would you mind having a look? Some fixes around the stream (and better testing :)

File: cmd/oaas/handler_build_test.go [hunk 1]
```diff
@@ -207,3 +211,110 @@ func TestHandleIncludedSourcesBadTypes(t *testing.T) {
...
+		// do anything)
+		expectedTimeSinceStart := time.Duration(i)*200*time.Millisecond + (100 * time.Millisecond)
+		assert.True(t, timeSinceStart < expectedTimeSinceStart, fmt.Sprintf("time since start %v bigger than expected %v", timeSinceStart, expectedTimeSinceStart))
+	}
+}
+
+func TestBuildErrorHandlingTar(t *testing.T) {
+	restore := main.MockOsbuildBinary(t, `#!/bin/sh
+
+# not creating an output dir, this will lead to errors from the "tar"
```
Comment (mvo5) [inline cmd/oaas/handler_build_test.go:303]: Not ideal, better would be to mock tar directly but I feel a bit lazy today :/

Comment (croissanne) [reply cmd/oaas/handler_build_test.go:303]: intimately familiar with that feeling

File: cmd/oaas/handler_build.go [hunk 2]
```diff
@@ -110,8 +88,10 @@ func runOsbuild(buildDir string, control *controlJSON, output io.Writer) (string
...
 	out, err := cmd.CombinedOutput()
 	if err != nil {
-		logrus.Errorf("Failed creating result tarball: %v", err)
-		return "", fmt.Errorf("cannot tar output directory: %v, output:\n%s", err, out)
+		err = fmt.Errorf("cannot tar output directory: %w, output:\n%s", err, out)
+		logrus.Errorf(err.Error())
+		mw.Write([]byte(err.Error()))
+		return "", err
 	}
 	logrus.Infof("tar output:\n%s", out)
```
Comment (croissanne) [inline cmd/oaas/handler_build.go:96]: I wonder, should we include the tar output in the build log?

**Review by croissanne — APPROVED** [review]
Thank you!

File: cmd/oaas/handler_build.go (hunk 2 above)
Comment (croissanne) [reply cmd/oaas/handler_build.go:96]: actually, osbuild needs to be full json, so let's keep the build log as just the osbuild output.

Comment (mvo5) [reply cmd/oaas/handler_build.go:96]: Thank you

**Pull request merged**